### PR TITLE
Add a note about the necessity to keep BuildEventArgsWriter AND StructuredLogViewer.BuildEventArgsWriter consistent

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
Fixes # https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/796

### Context
+ No special events for BuildCheckError/BuildCheckWarning are needed, these are handled as a part of https://github.com/dotnet/msbuild/blob/6dcd44ec025ef90ed5d6d832e7af49b6b2dc5b10/src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs#L482

### CODE REVIEW

File: src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs [hunk 1]
```diff
@@ -321,20 +321,6 @@ private BinaryLogRecordKind Write(BuildCheckResultMessage e)
             return BinaryLogRecordKind.BuildCheckMessage;
         }
 
-        private BinaryLogRecordKind Write(BuildCheckResultWarning e)
```
Comment (surayya-MS) [inline src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs:324]: why remove this?

Comment (YuliiaKovalova) [reply src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs:324]: It's handled as a part of https://github.com/dotnet/msbuild/blob/6dcd44ec025ef90ed5d6d832e7af49b6b2dc5b10/src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs#L482

We don't have any custom fields that need to be logged separately (yet)

Comment (surayya-MS) [reply src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs:324]: nit: add this to the description of the PR

**Review by surayya-MS — APPROVED** [review]

File: src/Build/Logging/BinaryLogger/BinaryLogger.cs [hunk 2]
```diff
@@ -79,6 +79,8 @@ public sealed class BinaryLogger : ILogger
         //    - new record kinds: BuildCheckMessageEvent, BuildCheckWarningEvent, BuildCheckErrorEvent,
         //    BuildCheckTracingEvent, BuildCheckAcquisitionEvent, BuildSubmissionStartedEvent
 
+        // MAKE SURE YOU KEEP BuildEventArgsWriter AND StructuredLogViewer.BuildEventArgsWriter IN SYNC WITH THE CHANGES ABOVE.
```
Comment (maridematte) [inline src/Build/Logging/BinaryLogger/BinaryLogger.cs:82]: I think giving a brief explanation on the why we need to do that would be nice here.

File: src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs (hunk 1 above)
Comment (JanKrivanek) [reply src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs:324]: However then those events will be deserialized as ordinary warnings.
They are still distinguishable manually - when one knows what to expect. But we cannot e.g. choose to skip them automatically during replay.

**Review by JanKrivanek — APPROVED** [review]

**Pull request merged**